plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy_test.go

```diff
@@ -175,6 +178,23 @@ func TestBootstrapClusterRoles(t *testing.T) {
 	testObjects(t, list, "cluster-roles.yaml")
 }
 
+func TestBootstrapClusterRolesWithFeatureGateEnabled(t *testing.T) {
+	featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.DynamicResourceAllocation, true)
+	list := &api.List{}
+	names := sets.NewString()
+	roles := map[string]runtime.Object{}
+	bootstrapRoles := bootstrappolicy.ClusterRoles()
+	for i := range bootstrapRoles {
+		role := bootstrapRoles[i]
+		names.Insert(role.Name)
+		roles[role.Name] = &role
+	}
+	for _, name := range names.List() {
+		list.Items = append(list.Items, roles[name])
+	}
+	testObjects(t, list, "cluster-roles-featuregates.yaml")
```

Comment:
Please also add a `map[string]string` that captures the `role name` `->` `cmp.Diff(regularRole, featureGateRole)` so it is easy to tell what the diff between the two is (roles with no diffs should be ignored).